### PR TITLE
Encryption (not working)

### PR DESCRIPTION
**Pull request by richajaindce**

### CODE REVIEW

File: ipa-core/src/report.rs [hunk 1]
```diff
@@ -522,17 +523,244 @@ where
...
+            self.site_domain(),
+        )
+        .unwrap(); // validated on construction
+        println!("constructed info :{:?}", info);
+
+        let mut ct_mk: GenericArray<u8, CTMKLength> =
+            *GenericArray::from_slice(self.mk_ciphertext());
+        println!("encrypted mk :{:?}", ct_mk);
+
+        let plaintext_mk = open_in_place(key_registry, self.encap_key_mk(), &mut ct_mk, &info)?;
```
Comment (richajaindce) [inline ipa-core/src/report.rs]: This is not working, HpkeError::OpenError is returned

File: ipa-core/src/report.rs [hunk 2]
```diff
@@ -631,4 +859,54 @@ mod test {
...
+        file.read_to_end(&mut bytes).expect("Error reading file");
+        let report = EncryptedMatchKeyReport::<_>::from_bytes(bytes.as_slice()).unwrap();
+
+        let result = report.decrypt(&key_registry);
+        println!("report is {:?}", result.unwrap());
+    }
+
+    #[test]
+    fn test_swift_decr_encryption() {
+        let mut rng = thread_rng();
```
Comment (richajaindce) [inline ipa-core/src/report.rs]: Trying to run this test function which returns an error. However, all it does is encrypt and decrypt a match key

File: ipa-core/src/hpke/mod.rs [hunk 3]
```diff
@@ -26,7 +26,7 @@ use crate::{
 
 /// IPA ciphersuite
 type IpaKem = hpke::kem::X25519HkdfSha256;
-type IpaAead = hpke::aead::AesGcm128;
+pub type IpaAead = hpke::aead::AesGcm128;
```
Comment (akoshelev) [inline ipa-core/src/hpke/mod.rs]: I think that is what is causing issues now - you're using AES-GCM-256 on Swift side and switched to AES-GCM-128. I don't know the reason, but I think it needs to be consistne

Comment (richajaindce) [reply ipa-core/src/hpke/mod.rs]: This did it! This was the last one remaining!!!

**Pull request closed without merging**